lms/djangoapps/grades/tasks.py

```diff
@@ -34,6 +35,14 @@ def recalculate_subsection_grade(user_id, course_id, usage_id, only_if_higher):
     course_key = CourseLocator.from_string(course_id)
     student = User.objects.get(id=user_id)
     scored_block_usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    if modified_time:
+        student_module = StudentModule.objects.get(
+            student_id=user_id,
+            module_state_key=scored_block_usage_key,
+            course_id=course_key,
+        )
```

Comment:
Problem, second part: This is using the `modified` value of the containing **subsection**, which was not updated as part of `set_score`.